great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -295,6 +298,72 @@ def get_batch_list_from_batch_request(
         self.sort_batches(batch_list)
         return batch_list
 
+    def _create_batch_spec_kwargs(self) -> dict[str, Any]:
+        """Creates batch_spec_kwargs used to instantiate a SqlAlchemyDatasourceBatchSpec
+
+        This is called by get_batch_list_from_batch_request to generate the batches.
+
+        Returns:
+            A dictionary that will be passed to SqlAlchemyDatasourceBatchSpec(**returned_dict)
+        """
+        raise NotImplementedError
+
+    def data_from(self) -> str:
+        """Returns the from clause that would be used to query this data
+
+        Returns:
+            The from clause which might be a table name or a query.
+        """
+        raise NotImplementedError
+
+
+class QueryAsset(SQLAsset):
```

Comment:
@billdirks -- I ❤️ the idea of the `SQLDataAsset` defining an interface for query, table, etc. to use!  Thanks!